json/shared/src/main/scala/fs2/data/json/selectors.scala

```diff
@@ -138,4 +161,12 @@ object IndexPredicate {
   case class Range(start: Int, end: Int) extends IndexPredicate {
     def apply(idx: Int): Boolean = start <= idx && idx <= end
   }
+
+  implicit val IndexPredicateShow: Show[IndexPredicate] = Show.show {
```

Comment:
Minor: For consistency, maybe use always `implicit object`?

Reply:
IIRC I did it this way because of some error of the typer complaining the implicit lacked explicit return type when looked up in the `Selector` one.